core/web/api/links.py

```diff
@@ -66,3 +66,51 @@ def multiupdate(self):
             updated.append(link.id)
 
         return render({"updated": updated})
+
+    @route("/", methods=['POST'])
+    @requires_permissions('write')
+    def new(self):
+        """Create a new link
+
+        Create a new link from the JSON object passed in the ``POST`` data.
+
+        :<json object params: JSON object containing object ids to link
+        """
+        params = request.json
+        type_src = params.pop('type_src', None)
+        type_dst = params.pop('type_dst', None)
+        db_objects_src = None
+        db_objects_dst = None
+
+        if type_src == 'observable':
```

Comment:
You can make the code cleaner by building a small dict that maps user-provided types with classes. e.g.

```
type_map = {
  'observable': observables.Observable,
  'entity': entities.Entity,
  'indicator': indicators.Indicator,
}
```

and then just do

```
db_objects_src  = type_map[type_src]
db_objects_dst  = type_map[type_dst]
```

Also, I think renaming `db_objects_src` to `src_object_class` throughout the function would be much clearer in this case.

Reply:
Indeed :)